In [304]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [305]:
# taking away sci notation
pd.set_option('display.float_format', lambda x: '%.5f' %x)

In [306]:
df_director = pd.read_csv('movie_director_wrtier.cvs')

In [307]:
df_director
# got the director for each for from sql data. 

Unnamed: 0   movie_id          director  \
0               0  tt0069049      Orson Welles   
1               1  tt0069049      Orson Welles   
2               2  tt0137204  Anthony Harrison   
3               3  tt0159369   John Mulholland   
4               4  tt0192528       Stuart Paul   
...           ...        ...               ...   
46815       77355  tt9825006    Rémi St-Michel   
46816       77356  tt9845398      Lucio Castro   
46817       77357  tt9852644       Loïc Darses   
46818       77358  tt9866344   Adib Alkhalidey   
46819       77359  tt9866344   Adib Alkhalidey   

                            primary_title                      genres  \
0              The Other Side of the Wind                       Drama   
1              The Other Side of the Wind                       Drama   
2                         Joe Finds Grace  Adventure,Animation,Comedy   
3      Cooper and Hemingway: The True Gen                 Documentary   
4                           Heaven & Hell                       Drama   
...                                   ...                         ...   
46815                 Avant qu'on explose                      Comedy   
46816                  End of the Century                       Drama   
46817                   La Fin des terres                 Documentary   
46818                       Mon Ami Walid                Comedy,Drama   
46819                       Mon Ami Walid                Comedy,Drama   

       start_year  averagerating region            writers  
0            2018        6.90000     US          Oja Kodar  
1            2018        6.90000     US       Orson Welles  
2            2017        8.10000     CA   Anthony Harrison  
3            2013        7.60000     US    John Mulholland  
4            2018        4.00000     US        Stuart Paul  
...           ...            ...    ...                ...  
46815        2019        6.90000     CA  Eric K. Boulianne  
46816        2019        7.20000     US       Lucio Castro  
46817        2019        7.00000     CA        Loïc Darses  
46818        2019        5.70000     CA    Adib Alkhalidey  
46819        2019        5.70000     CA     Julien Lacroix  

[46820 rows x 9 columns]

In [308]:
profit_df = pd.read_csv('world_profit_per_six_genres.csv')

In [309]:
profit_df
# got profit_df from the question 2 file

Unnamed: 0                   title  start_year  runtime_minutes  \
0           1342            Dark Phoenix        2019        113.00000   
1           1343            Dark Phoenix        2019        113.00000   
2           1344            Dark Phoenix        2019        113.00000   
3           3351         Mars Needs Moms        2011         88.00000   
4           3349         Mars Needs Moms        2011         88.00000   
...          ...                     ...         ...              ...   
1833        2892          Jurassic World        2015        124.00000   
1834         535  Avengers: Infinity War        2018        149.00000   
1835         534  Avengers: Infinity War        2018        149.00000   
1836         533  Avengers: Infinity War        2018        149.00000   
1837        6538                 Titanic        2018        115.00000   

         genres  id  release_date  production_budget  domestic_gross  \
0        Action   3   Jun 7, 2019    350000000.00000  42762350.00000   
1     Adventure   3   Jun 7, 2019    350000000.00000  42762350.00000   
2        Sci-Fi   3   Jun 7, 2019    350000000.00000  42762350.00000   
3        Family  94  Mar 11, 2011    150000000.00000  21392758.00000   
4     Adventure  94  Mar 11, 2011    150000000.00000  21392758.00000   
...         ...  ..           ...                ...             ...   
1833  Adventure  34  Jun 12, 2015    215000000.00000 652270625.00000   
1834     Sci-Fi   7  Apr 27, 2018    300000000.00000 678815482.00000   
1835  Adventure   7  Apr 27, 2018    300000000.00000 678815482.00000   
1836     Action   7  Apr 27, 2018    300000000.00000 678815482.00000   
1837     Family  43  Dec 19, 1997    200000000.00000 659363944.00000   

      worldwide_gross     world_profit  
0     149762350.00000 -200237650.00000  
1     149762350.00000 -200237650.00000  
2     149762350.00000 -200237650.00000  
3      39549758.00000 -110450242.00000  
4      39549758.00000 -110450242.00000  
...               ...              ...  
1833 1648854864.00000 1433854864.00000  
1834 2048134200.00000 1748134200.00000  
1835 2048134200.00000 1748134200.00000  
1836 2048134200.00000 1748134200.00000  
1837 2208208395.00000 2008208395.00000  

[1838 rows x 11 columns]

In [310]:
#dropping columns
profit_df = profit_df.drop('Unnamed: 0', axis=1) 
df_director = df_director.drop('Unnamed: 0', axis=1) 

In [311]:
# checking drop
profit_df.head()

title  start_year  runtime_minutes     genres  id  release_date  \
0     Dark Phoenix        2019        113.00000     Action   3   Jun 7, 2019   
1     Dark Phoenix        2019        113.00000  Adventure   3   Jun 7, 2019   
2     Dark Phoenix        2019        113.00000     Sci-Fi   3   Jun 7, 2019   
3  Mars Needs Moms        2011         88.00000     Family  94  Mar 11, 2011   
4  Mars Needs Moms        2011         88.00000  Adventure  94  Mar 11, 2011   

   production_budget  domestic_gross  worldwide_gross     world_profit  
0    350000000.00000  42762350.00000  149762350.00000 -200237650.00000  
1    350000000.00000  42762350.00000  149762350.00000 -200237650.00000  
2    350000000.00000  42762350.00000  149762350.00000 -200237650.00000  
3    150000000.00000  21392758.00000   39549758.00000 -110450242.00000  
4    150000000.00000  21392758.00000   39549758.00000 -110450242.00000

In [312]:
# checking drop
df_director.head()

movie_id          director                       primary_title  \
0  tt0069049      Orson Welles          The Other Side of the Wind   
1  tt0069049      Orson Welles          The Other Side of the Wind   
2  tt0137204  Anthony Harrison                     Joe Finds Grace   
3  tt0159369   John Mulholland  Cooper and Hemingway: The True Gen   
4  tt0192528       Stuart Paul                       Heaven & Hell   

                       genres  start_year  averagerating region  \
0                       Drama        2018        6.90000     US   
1                       Drama        2018        6.90000     US   
2  Adventure,Animation,Comedy        2017        8.10000     CA   
3                 Documentary        2013        7.60000     US   
4                       Drama        2018        4.00000     US   

            writers  
0         Oja Kodar  
1      Orson Welles  
2  Anthony Harrison  
3   John Mulholland  
4       Stuart Paul

In [313]:
#dropping unused/un needed columns in df_director
df_director = df_director.drop(['genres', 'start_year', 'movie_id', 'region'], axis=1)

In [314]:
# checking the drop
df_director.head()

director                       primary_title  averagerating  \
0      Orson Welles          The Other Side of the Wind        6.90000   
1      Orson Welles          The Other Side of the Wind        6.90000   
2  Anthony Harrison                     Joe Finds Grace        8.10000   
3   John Mulholland  Cooper and Hemingway: The True Gen        7.60000   
4       Stuart Paul                       Heaven & Hell        4.00000   

            writers  
0         Oja Kodar  
1      Orson Welles  
2  Anthony Harrison  
3   John Mulholland  
4       Stuart Paul

In [315]:
# joining director with profit columns with a new variable df
df = df_director.set_index('primary_title').join(profit_df.set_index('title'), how='inner')

In [316]:
# checking the join
df

director  averagerating           writers  \
12 Strong                   Nicolai Fuglsig        6.60000         Ted Tally   
12 Strong                   Nicolai Fuglsig        6.60000       Peter Craig   
12 Strong                   Nicolai Fuglsig        6.60000      Doug Stanton   
127 Hours                       Danny Boyle        7.60000       Danny Boyle   
127 Hours                       Danny Boyle        7.60000     Simon Beaufoy   
...                                     ...            ...               ...   
Zootopia                       Byron Howard        8.00000      Jennifer Lee   
xXx: Return of Xander Cage      D.J. Caruso        5.20000       Rich Wilkes   
xXx: Return of Xander Cage      D.J. Caruso        5.20000       Rich Wilkes   
xXx: Return of Xander Cage      D.J. Caruso        5.20000  F. Scott Frazier   
xXx: Return of Xander Cage      D.J. Caruso        5.20000  F. Scott Frazier   

                            start_year  runtime_minutes     genres  id  \
12 Strong                         2018        130.00000     Action  64   
12 Strong                         2018        130.00000     Action  64   
12 Strong                         2018        130.00000     Action  64   
127 Hours                         2010         94.00000  Adventure   6   
127 Hours                         2010         94.00000  Adventure   6   
...                                ...              ...        ...  ..   
Zootopia                          2016        108.00000  Adventure  57   
xXx: Return of Xander Cage        2017        107.00000     Action  15   
xXx: Return of Xander Cage        2017        107.00000  Adventure  15   
xXx: Return of Xander Cage        2017        107.00000     Action  15   
xXx: Return of Xander Cage        2017        107.00000  Adventure  15   

                            release_date  production_budget  domestic_gross  \
12 Strong                   Jan 19, 2018     35000000.00000  45819713.00000   
12 Strong                   Jan 19, 2018     35000000.00000  45819713.00000   
12 Strong                   Jan 19, 2018     35000000.00000  45819713.00000   
127 Hours                    Nov 5, 2010     18000000.00000  18335230.00000   
127 Hours                    Nov 5, 2010     18000000.00000  18335230.00000   
...                                  ...                ...             ...   
Zootopia                     Mar 4, 2016    150000000.00000 341268248.00000   
xXx: Return of Xander Cage  Jan 20, 2017     85000000.00000  44898413.00000   
xXx: Return of Xander Cage  Jan 20, 2017     85000000.00000  44898413.00000   
xXx: Return of Xander Cage  Jan 20, 2017     85000000.00000  44898413.00000   
xXx: Return of Xander Cage  Jan 20, 2017     85000000.00000  44898413.00000   

                            worldwide_gross    world_profit  
12 Strong                    71118378.00000  36118378.00000  
12 Strong                    71118378.00000  36118378.00000  
12 Strong                    71118378.00000  36118378.00000  
127 Hours                    60217171.00000  42217171.00000  
127 Hours                    60217171.00000  42217171.00000  
...                                     ...             ...  
Zootopia                   1019429616.00000 869429616.00000  
xXx: Return of Xander Cage  345033359.00000 260033359.00000  
xXx: Return of Xander Cage  345033359.00000 260033359.00000  
xXx: Return of Xander Cage  345033359.00000 260033359.00000  
xXx: Return of Xander Cage  345033359.00000 260033359.00000  

[5560 rows x 12 columns]

In [317]:
# reset index 
df.reset_index(inplace = True)

In [319]:
# checking
df.head()

index         director  averagerating        writers  start_year  \
0  12 Strong  Nicolai Fuglsig        6.60000      Ted Tally        2018   
1  12 Strong  Nicolai Fuglsig        6.60000    Peter Craig        2018   
2  12 Strong  Nicolai Fuglsig        6.60000   Doug Stanton        2018   
3  127 Hours      Danny Boyle        7.60000    Danny Boyle        2010   
4  127 Hours      Danny Boyle        7.60000  Simon Beaufoy        2010   

   runtime_minutes     genres  id  release_date  production_budget  \
0        130.00000     Action  64  Jan 19, 2018     35000000.00000   
1        130.00000     Action  64  Jan 19, 2018     35000000.00000   
2        130.00000     Action  64  Jan 19, 2018     35000000.00000   
3         94.00000  Adventure   6   Nov 5, 2010     18000000.00000   
4         94.00000  Adventure   6   Nov 5, 2010     18000000.00000   

   domestic_gross  worldwide_gross   world_profit  
0  45819713.00000   71118378.00000 36118378.00000  
1  45819713.00000   71118378.00000 36118378.00000  
2  45819713.00000   71118378.00000 36118378.00000  
3  18335230.00000   60217171.00000 42217171.00000  
4  18335230.00000   60217171.00000 42217171.00000

In [320]:
df = df.drop(['id','production_budget','domestic_gross','worldwide_gross','release_date'], axis=1)
# dropping unneed columns

In [321]:
# renaming the index column to title
df.rename(columns = {'index' : 'title'}, inplace=True)

In [324]:
# checking
df

title         director  averagerating  \
0                      12 Strong  Nicolai Fuglsig        6.60000   
1                      12 Strong  Nicolai Fuglsig        6.60000   
2                      12 Strong  Nicolai Fuglsig        6.60000   
3                      127 Hours      Danny Boyle        7.60000   
4                      127 Hours      Danny Boyle        7.60000   
...                          ...              ...            ...   
5555                    Zootopia     Byron Howard        8.00000   
5556  xXx: Return of Xander Cage      D.J. Caruso        5.20000   
5557  xXx: Return of Xander Cage      D.J. Caruso        5.20000   
5558  xXx: Return of Xander Cage      D.J. Caruso        5.20000   
5559  xXx: Return of Xander Cage      D.J. Caruso        5.20000   

               writers  start_year  runtime_minutes     genres    world_profit  
0            Ted Tally        2018        130.00000     Action  36118378.00000  
1          Peter Craig        2018        130.00000     Action  36118378.00000  
2         Doug Stanton        2018        130.00000     Action  36118378.00000  
3          Danny Boyle        2010         94.00000  Adventure  42217171.00000  
4        Simon Beaufoy        2010         94.00000  Adventure  42217171.00000  
...                ...         ...              ...        ...             ...  
5555      Jennifer Lee        2016        108.00000  Adventure 869429616.00000  
5556       Rich Wilkes        2017        107.00000     Action 260033359.00000  
5557       Rich Wilkes        2017        107.00000  Adventure 260033359.00000  
5558  F. Scott Frazier        2017        107.00000     Action 260033359.00000  
5559  F. Scott Frazier        2017        107.00000  Adventure 260033359.00000  

[5560 rows x 8 columns]

In [325]:
# checking for duplicates
dup = df[df.duplicated()]
print(len(dup))

0


In [326]:
# indexing for only Animation genere, groubying by the director and 
# writer for average of world_profit and average rating of the movie

ani_profit = df[df['genres'] == 'Animation'].groupby(['director','writers']).mean().sort_values(by = ['world_profit','averagerating'], ascending = False)

In [328]:
ani_profit.shape
# getting the shape of dataframe

(496, 4)

In [338]:
ani_profit.iloc[:20]


averagerating  start_year  \
director     writers                                              
Chris Buck   Chris Buck                     7.50000  2013.00000   
             Hans Christian Andersen        7.50000  2013.00000   
             Jennifer Lee                   7.50000  2013.00000   
             Shane Morris                   7.50000  2013.00000   
Adam Green   Adam Green                     6.20000  2013.00000   
Kyle Balda   Brian Lynch                    6.40000  2015.00000   
Brad Bird    Brad Bird                      7.70000  2018.00000   
Kyle Balda   Cinco Paul                     6.30000  2017.00000   
             Ken Daurio                     6.30000  2017.00000   
Byron Howard Byron Howard                   8.00000  2016.00000   
             Jared Bush                     8.00000  2016.00000   
             Jennifer Lee                   8.00000  2016.00000   
             Jim Reardon                    8.00000  2016.00000   
             Josie Trinidad                 8.00000  2016.00000   
             Kellie D. Lewis                8.00000  2016.00000   
             Phil Johnston                  8.00000  2016.00000   
             Rich Moore                     8.00000  2016.00000   
Lee Unkrich  Andrew Stanton                 8.30000  2010.00000   
             John Lasseter                  8.30000  2010.00000   
             Michael Arndt                  8.30000  2010.00000   

                                      runtime_minutes     world_profit  
director     writers                                                    
Chris Buck   Chris Buck                     102.00000 1122469910.00000  
             Hans Christian Andersen        102.00000 1122469910.00000  
             Jennifer Lee                   102.00000 1122469910.00000  
             Shane Morris                   102.00000 1122469910.00000  
Adam Green   Adam Green                     102.00000 1122469910.00000  
Kyle Balda   Brian Lynch                     91.00000 1086336173.00000  
Brad Bird    Brad Bird                      118.00000 1042520711.00000  
Kyle Balda   Cinco Paul                      89.00000  959727750.00000  
             Ken Daurio                      89.00000  959727750.00000  
Byron Howard Byron Howard                   108.00000  869429616.00000  
             Jared Bush                     108.00000  869429616.00000  
             Jennifer Lee                   108.00000  869429616.00000  
             Jim Reardon                    108.00000  869429616.00000  
             Josie Trinidad                 108.00000  869429616.00000  
             Kellie D. Lewis                108.00000  869429616.00000  
             Phil Johnston                  108.00000  869429616.00000  
             Rich Moore                     108.00000  869429616.00000  
Lee Unkrich  Andrew Stanton                 103.00000  868879522.00000  
             John Lasseter                  103.00000  868879522.00000  
             Michael Arndt                  103.00000  868879522.00000

# Notes

Jennifer Lee (writer) is in the top 7 movies by avg_profit in the Animation genre
wrote 2 movie in 2013 with Chris Buck and 2016 Byron Howard 
both movies had a avg_rating above 7 (7.5 - 2013 and 8.0- 2016) was on the writer for Frozen (Chris Buck director)


Adam Green directed and was the only writer of his moive who is top 2nd avg_profit wise

Brad Bird directed and was the only writer of his moive who is top 4th avg_profit wise

# Recommendation

All these dire